R/etwfe.R

```diff
@@ -219,8 +226,19 @@ etwfe = function(
   
   ## Estimation ----
   
-  ## Full formula
-  Fml = Formula::as.Formula(paste(lhs, " ~ ", rhs, "|", fes)) 
+  ##### NEW: demean the interacted variable ######
+  if (!is.null(intvar)){
+    intvar_dm <- data[intvar] - mean(dplyr::pull(dplyr::distinct(dplyr::filter(data, .Dtreat == T), get(id), .keep_all = T), intvar), na.rm = T)
+    intvar_dm_df = stats::setNames(intvar_dm, paste0(intvar, "_dm")) # give a name
+    data = cbind(data, intvar_dm_df)
+  } 
+  
+  ## NEW: Full formula with interaction
+  if( !is.null(intvar) ) {
+    Fml = Formula::as.Formula(paste(lhs, " ~ ", rhs, ":", paste0(intvar, "_dm"), "+", rhs, "+ i(", gvar, ", as_factor(", tvar, ")):", intvar, "|", fes))
```

Comment:
As per my earlier comment, please use the idiomatic `i(gvar, i.tvar)` syntax instead of `as_factor`. You may need to bring intvar (xvar) to the front, so that it's written as `intvar:i(gvar, i.tvar)`.  Also: what's happening to `gref`, `tref`, etc. here? Have they dropped out by mistake?

You'll ultimately need to test with `devtools::check` or `tinytest::run_build_dir("inst/tinytest")`.